pkg/sql/schemachanger/schemachanger_test: TestAlterTableDMLInjection failed

pkg/sql/schemachanger/schemachanger_test.TestAlterTableDMLInjection [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12072200?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12072200?buildTab=artifacts#/) on master @ [c9a244db4eeddc7163d67c1bf28a15ed218d633e](https://github.com/cockroachdb/cockroach/commits/c9a244db4eeddc7163d67c1bf28a15ed218d633e):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestAlterTableDMLInjection (14m55s)
	TestAlterTableDMLInjection/drop_constraint_check (14m55s)

```
Stack: 

```
goroutine 279792 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestAlterTableDMLInjection/drop_partial_index
=== PAUSE TestAlterTableDMLInjection/drop_partial_index
=== CONT  TestAlterTableDMLInjection/drop_partial_index
=== RUN   TestAlterTableDMLInjection/add_column_stored
=== PAUSE TestAlterTableDMLInjection/add_column_stored
=== CONT  TestAlterTableDMLInjection/add_column_stored
=== NAME  TestAlterTableDMLInjection/create_partial_index
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/add_constraint_check_udf
=== PAUSE TestAlterTableDMLInjection/add_constraint_check_udf
=== CONT  TestAlterTableDMLInjection/add_constraint_check_udf
=== RUN   TestAlterTableDMLInjection/add_constraint_foreign_key
=== PAUSE TestAlterTableDMLInjection/add_constraint_foreign_key
=== CONT  TestAlterTableDMLInjection/add_constraint_foreign_key
=== NAME  TestAlterTableDMLInjection/create_materialized_view_from_index
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/add_constraint_unique_without_index
=== PAUSE TestAlterTableDMLInjection/add_constraint_unique_without_index
=== CONT  TestAlterTableDMLInjection/add_constraint_unique_without_index
=== RUN   TestAlterTableDMLInjection/create_index
=== PAUSE TestAlterTableDMLInjection/create_index
=== CONT  TestAlterTableDMLInjection/create_index
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== NAME  TestAlterTableDMLInjection/drop_constraint_check
    dml_injection_test.go:500: 
        	Error Trace:	pkg/sql/schemachanger/schemachanger_test_test/pkg/sql/schemachanger/dml_injection_test.go:500
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:189
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:139
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scdeps/run_deps.go:144
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1670
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1759
        	            				github.com/cockroachdb/cockroach/pkg/kv/txn.go:1012
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1055
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1018
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:993
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1746
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1668
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scdeps/run_deps.go:109
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:135
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scjob/job.go:118
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scjob/job.go:45
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1666
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1667
        	            				github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:477
        	            				github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:389
        	            				github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
        	            				src/runtime/asm_arm64.s:1172
        	Error:      	[][]string{[]string{"PreCommitPhase:1", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"PreCommitPhase:1", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"PreCommitPhase:1", "PreCommitPhase:2", "update", "2"}, []string{"PreCommitPhase:1", "n/a", "n/a", "1"}, []string{"PreCommitPhase:2", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"PreCommitPhase:2", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"PreCommitPhase:2", "n/a", "n/a", "1"}, []string{"StatementPhase:1", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"StatementPhase:1", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"StatementPhase:1", "PreCommitPhase:1", "update", "2"}, []string{"StatementPhase:1", "PreCommitPhase:2", "update", "2"}, []string{"StatementPhase:1", "n/a", "n/a", "1"}, []string{"pre-schema-change", "n/a", "n/a", "1"}} does not contain []string{"PostCommitNonRevertiblePhase:1", "n/a", "n/a", "1"}
        	Test:       	TestAlterTableDMLInjection/drop_constraint_check
        	Messages:   	phaseOrdinal=PostCommitNonRevertiblePhase:1
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTableDMLInjection.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32108

pkg/sql/schemachanger/schemachanger_test.TestAlterTableDMLInjection [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12089374?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12089374?buildTab=artifacts#/) on master @ [d26a180f914dbb0a727566f1ac1f4f3ee4c624d9](https://github.com/cockroachdb/cockroach/commits/d26a180f914dbb0a727566f1ac1f4f3ee4c624d9):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestAlterTableDMLInjection (14m55s)
	TestAlterTableDMLInjection/drop_constraint_foreign_key (14m55s)

```
Stack: 

```
goroutine 198344 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestAlterTableDMLInjection
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/3e4c90800e6eae84b5bf83507e7ef6e8/logTestAlterTableDMLInjection3918355023
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestAlterTableDMLInjection/add_column_stored
=== PAUSE TestAlterTableDMLInjection/add_column_stored
=== CONT  TestAlterTableDMLInjection/add_column_stored
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/drop_column_with_index
=== PAUSE TestAlterTableDMLInjection/drop_column_with_index
=== CONT  TestAlterTableDMLInjection/drop_column_with_index
=== RUN   TestAlterTableDMLInjection/drop_column_with_unique_index
=== PAUSE TestAlterTableDMLInjection/drop_column_with_unique_index
=== CONT  TestAlterTableDMLInjection/drop_column_with_unique_index
=== RUN   TestAlterTableDMLInjection/drop_expression_index
=== PAUSE TestAlterTableDMLInjection/drop_expression_index
=== CONT  TestAlterTableDMLInjection/drop_expression_index
=== RUN   TestAlterTableDMLInjection/drop_index_using_hash
=== PAUSE TestAlterTableDMLInjection/drop_index_using_hash
=== CONT  TestAlterTableDMLInjection/drop_index_using_hash
=== RUN   TestAlterTableDMLInjection/add_column_stored_family
=== PAUSE TestAlterTableDMLInjection/add_column_stored_family
=== CONT  TestAlterTableDMLInjection/add_column_stored_family
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/alter_primary_key_using_columns_using_hash
=== PAUSE TestAlterTableDMLInjection/alter_primary_key_using_columns_using_hash
=== CONT  TestAlterTableDMLInjection/alter_primary_key_using_columns_using_hash
    skip.go:201: [https://github.com/cockroachdb/cockroach/issues/111570]
=== RUN   TestAlterTableDMLInjection/create_index_using_hash
=== PAUSE TestAlterTableDMLInjection/create_index_using_hash
=== CONT  TestAlterTableDMLInjection/create_index_using_hash
    skip.go:201: [https://github.com/cockroachdb/cockroach/issues/111621]
=== RUN   TestAlterTableDMLInjection/drop_column_stored
=== PAUSE TestAlterTableDMLInjection/drop_column_stored
=== CONT  TestAlterTableDMLInjection/drop_column_stored
=== RUN   TestAlterTableDMLInjection/drop_constraint_unique_without_index
=== PAUSE TestAlterTableDMLInjection/drop_constraint_unique_without_index
=== CONT  TestAlterTableDMLInjection/drop_constraint_unique_without_index
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTableDMLInjection.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestAlterTableDMLInjection [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12102811?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12102811?buildTab=artifacts#/) on master @ [adb6ce6526cec2b9633ef4819e543747cce31ad0](https://github.com/cockroachdb/cockroach/commits/adb6ce6526cec2b9633ef4819e543747cce31ad0):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestAlterTableDMLInjection (14m55s)
	TestAlterTableDMLInjection/drop_constraint_foreign_key (14m46s)

```
Stack: 

```
goroutine 244164 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== PAUSE TestAlterTableDMLInjection/add_constraint_foreign_key
=== CONT  TestAlterTableDMLInjection/add_constraint_foreign_key
=== RUN   TestAlterTableDMLInjection/add_primary_key
=== PAUSE TestAlterTableDMLInjection/add_primary_key
=== CONT  TestAlterTableDMLInjection/add_primary_key
=== RUN   TestAlterTableDMLInjection/alter_primary_key_using_columns
=== PAUSE TestAlterTableDMLInjection/alter_primary_key_using_columns
=== CONT  TestAlterTableDMLInjection/alter_primary_key_using_columns
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/create_unique_index
=== PAUSE TestAlterTableDMLInjection/create_unique_index
=== CONT  TestAlterTableDMLInjection/create_unique_index
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/create_index
=== PAUSE TestAlterTableDMLInjection/create_index
=== CONT  TestAlterTableDMLInjection/create_index
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/drop_column_with_index
=== PAUSE TestAlterTableDMLInjection/drop_column_with_index
=== CONT  TestAlterTableDMLInjection/drop_column_with_index
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/drop_constraint_foreign_key
=== PAUSE TestAlterTableDMLInjection/drop_constraint_foreign_key
=== CONT  TestAlterTableDMLInjection/drop_constraint_foreign_key
    dml_injection_test.go:500: 
        	Error Trace:	pkg/sql/schemachanger/schemachanger_test_test/pkg/sql/schemachanger/dml_injection_test.go:500
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:189
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:139
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scdeps/run_deps.go:144
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1670
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1759
        	            				github.com/cockroachdb/cockroach/pkg/kv/txn.go:1012
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1055
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1018
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:993
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1746
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1668
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scdeps/run_deps.go:109
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:135
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scjob/job.go:118
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scjob/job.go:45
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1668
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1669
        	            				github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:477
        	            				github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:389
        	            				github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
        	            				src/runtime/asm_arm64.s:1172
        	Error:      	[][]string{[]string{"PreCommitPhase:1", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"PreCommitPhase:1", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"PreCommitPhase:1", "PreCommitPhase:2", "update", "2"}, []string{"PreCommitPhase:1", "n/a", "n/a", "1"}, []string{"PreCommitPhase:2", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"PreCommitPhase:2", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"PreCommitPhase:2", "n/a", "n/a", "1"}, []string{"StatementPhase:1", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"StatementPhase:1", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"StatementPhase:1", "PreCommitPhase:1", "update", "2"}, []string{"StatementPhase:1", "PreCommitPhase:2", "update", "2"}, []string{"StatementPhase:1", "n/a", "n/a", "1"}, []string{"pre-schema-change", "n/a", "n/a", "1"}} does not contain []string{"PostCommitNonRevertiblePhase:1", "n/a", "n/a", "1"}
        	Test:       	TestAlterTableDMLInjection/drop_constraint_foreign_key
        	Messages:   	phaseOrdinal=PostCommitNonRevertiblePhase:1
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTableDMLInjection.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/schemachanger/schemachanger_test.TestAlterTableDMLInjection [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12136185?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12136185?buildTab=artifacts#/) on master @ [1309e323927461efc6fab8da6a7fb4b996a95fdb](https://github.com/cockroachdb/cockroach/commits/1309e323927461efc6fab8da6a7fb4b996a95fdb):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestAlterTableDMLInjection (14m55s)
	TestAlterTableDMLInjection/drop_constraint_foreign_key (14m55s)

```
Stack: 

```
goroutine 234630 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== CONT  TestAlterTableDMLInjection/drop_column_with_partial_index
    skip.go:201: [https://github.com/cockroachdb/cockroach/issues/97813]
=== RUN   TestAlterTableDMLInjection/drop_constraint_foreign_key
=== PAUSE TestAlterTableDMLInjection/drop_constraint_foreign_key
=== CONT  TestAlterTableDMLInjection/drop_constraint_foreign_key
    dml_injection_test.go:500: 
        	Error Trace:	pkg/sql/schemachanger/schemachanger_test_test/pkg/sql/schemachanger/dml_injection_test.go:500
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:189
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:139
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scdeps/run_deps.go:144
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1673
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1762
        	            				github.com/cockroachdb/cockroach/pkg/kv/txn.go:1012
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1055
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:1018
        	            				github.com/cockroachdb/cockroach/pkg/kv/db.go:993
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1749
        	            				github.com/cockroachdb/cockroach/pkg/sql/internal.go:1671
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scdeps/run_deps.go:109
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scrun/scrun.go:135
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scjob/job.go:118
        	            				github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scjob/job.go:45
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1668
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1669
        	            				github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:477
        	            				github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:389
        	            				github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:484
        	            				src/runtime/asm_arm64.s:1172
        	Error:      	[][]string{[]string{"PreCommitPhase:1", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"PreCommitPhase:1", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"PreCommitPhase:1", "PreCommitPhase:2", "update", "2"}, []string{"PreCommitPhase:1", "n/a", "n/a", "1"}, []string{"PreCommitPhase:2", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"PreCommitPhase:2", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"PreCommitPhase:2", "n/a", "n/a", "1"}, []string{"StatementPhase:1", "PostCommitNonRevertiblePhase:1", "delete", "1"}, []string{"StatementPhase:1", "PostCommitNonRevertiblePhase:1", "update", "1"}, []string{"StatementPhase:1", "PreCommitPhase:1", "update", "2"}, []string{"StatementPhase:1", "PreCommitPhase:2", "update", "2"}, []string{"StatementPhase:1", "n/a", "n/a", "1"}, []string{"pre-schema-change", "n/a", "n/a", "1"}} does not contain []string{"PostCommitNonRevertiblePhase:1", "n/a", "n/a", "1"}
        	Test:       	TestAlterTableDMLInjection/drop_constraint_foreign_key
        	Messages:   	phaseOrdinal=PostCommitNonRevertiblePhase:1
=== RUN   TestAlterTableDMLInjection/validate_constraint
=== PAUSE TestAlterTableDMLInjection/validate_constraint
=== CONT  TestAlterTableDMLInjection/validate_constraint
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/add_column_virtual
=== PAUSE TestAlterTableDMLInjection/add_column_virtual
=== CONT  TestAlterTableDMLInjection/add_column_virtual
=== RUN   TestAlterTableDMLInjection/alter_primary_key_using_columns_using_hash
=== PAUSE TestAlterTableDMLInjection/alter_primary_key_using_columns_using_hash
=== CONT  TestAlterTableDMLInjection/alter_primary_key_using_columns_using_hash
    skip.go:201: [https://github.com/cockroachdb/cockroach/issues/111570]
=== RUN   TestAlterTableDMLInjection/alter_primary_key_using_columns
=== PAUSE TestAlterTableDMLInjection/alter_primary_key_using_columns
=== CONT  TestAlterTableDMLInjection/alter_primary_key_using_columns
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
=== RUN   TestAlterTableDMLInjection/create_partial_index
=== PAUSE TestAlterTableDMLInjection/create_partial_index
=== CONT  TestAlterTableDMLInjection/create_partial_index
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterTableDMLInjection.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/schemachanger/dml_injection_test.go
+++ b/pkg/sql/schemachanger/dml_injection_test.go
@@ -303,6 +303,7 @@ func TestAlterTableDMLInjection(t *testing.T) {
 		{
 			desc:         "create index",
 			schemaChange: "CREATE INDEX idx ON tbl (val)",
+			skipIssue:    112421,
 		},
 		{
 			desc:         "drop index",
@@ -361,6 +362,7 @@ func TestAlterTableDMLInjection(t *testing.T) {
 			desc:         "drop partial index",
 			setup:        []string{"CREATE INDEX idx ON tbl (val) WHERE val > 1"},
 			schemaChange: "DROP INDEX idx",
+			skipIssue:    112417,
 		},
 		{
 			desc: "drop column with partial index",
@@ -400,6 +402,7 @@ func TestAlterTableDMLInjection(t *testing.T) {
 				"CREATE MATERIALIZED VIEW mv AS SELECT * FROM tbl@idx",
 			},
 			schemaChange: "DROP INDEX idx CASCADE",
+			skipIssue:    112418,
 		},
 	}
 

